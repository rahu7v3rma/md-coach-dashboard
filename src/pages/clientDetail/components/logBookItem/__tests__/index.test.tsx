import { RenderResult, fireEvent, render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import LogBookItem from '..';

const props = {
    children: <>children</>,
    icon: 'icon',
    title: 'title',
    time: '00:00',
    width: '100px',
    height: '100px',
    image: 'image',
    onClick: jest.fn()
};

jest.mock('src/navigation');
global.URL.createObjectURL = jest.fn().mockImplementation(() => props.image);

let view!: RenderResult;
beforeEach(async () => {
    await act(() => (view = render(<LogBookItem {...props} />)));
    const foodImage = view.container.querySelector(`img[src="${props.image}"]`);
    await act(() => fireEvent.load(foodImage as HTMLElement));
});

test('Snapshot', async () => {
    expect(view.asFragment()).toMatchSnapshot();
});

test('renders LogBookItem component with all props and an image', async () => {
    const card = view.container.firstElementChild;
    const [header, body] = card!.children;
    const [titleContainer, foodImage, timeContainer] = header.children;
    const [titleIcon, titleText] = titleContainer.children;
    expect(card).toHaveStyle(`width:${props.width};height:${props.height};`);
    expect(titleIcon.getAttribute('src')).toBe(props.icon);
    expect(titleText.textContent).toBe(props.title);
    expect(foodImage.firstElementChild?.getAttribute('src')).toBe(props.image);
    expect(timeContainer.firstElementChild?.textContent).toBe(props.time);
    expect(body.textContent).toBe('children');
});

test('renders LogBookItem component without an image', async () => {
    act(() => view.rerender(<LogBookItem {...props} image={undefined} />));
    const foodImage = view.container.querySelector(`img[src="${props.image}"]`);
    expect(foodImage).toBeNull();
});

test('handles click event with the onClick prop', async () => {
    const card = view.container.firstElementChild;
    await act(() => fireEvent.click(card as HTMLElement));
    expect(props.onClick).toBeCalled();
});

test('handles image modal when an image is clicked, closes image modal when the modal is closed', async () => {
    let modalContent = view.container.querySelector('#modalContent');
    expect(modalContent).toBeNull();
    const foodImage = view.container.querySelector(`img[src="${props.image}"]`);
    await act(() => fireEvent.click(foodImage?.parentElement as HTMLElement));
    modalContent = view.container.querySelector('#modalContent');
    expect(modalContent).toBeVisible();
    const modalCloseButton = view.container.querySelector('#modalCloseButton');
    await act(() => fireEvent.click(modalCloseButton as HTMLElement));
    modalContent = view.container.querySelector('#modalContent');
    expect(modalContent).toBeNull();
});
