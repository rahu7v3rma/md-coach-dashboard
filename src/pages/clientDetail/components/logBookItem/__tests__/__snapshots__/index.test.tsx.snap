// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-ispOId eMkvam"
    height="100px"
    width="100px"
  >
    <div
      class="sc-koqlAI bnSEwg"
    >
      <div
        class="sc-kGRGSO leolIO"
      >
        <img
          class="sc-hKhKOZ oGqhD"
          src="icon"
        />
        <p
          class="sc-jJTsDX lmkThR"
          color="#72996B"
          font-family="\\"Poppins\\", sans-serif"
          font-size="14"
          font-weight="600"
        >
          title
        </p>
      </div>
      <div>
        <img
          alt=""
          class="sc-fAMvPz bpvgxV"
          src="image"
        />
      </div>
      <div
        class="sc-kGRGSO sc-jDnxRY leolIO"
      >
        <p
          class="sc-jJTsDX UehbF"
          color="#271A51"
          font-family="\\"Poppins\\", sans-serif"
          font-size="14"
          font-weight="500"
        >
          00:00
        </p>
      </div>
    </div>
    <div
      class="sc-grzvtn"
    >
      children
    </div>
  </div>
</DocumentFragment>
`;
